manifests/prepare/package.pp

```diff
@@ -28,12 +38,41 @@
     $mode = '0644'
   }
 
-  file { $local_package_file_path:
-    ensure  => file,
-    owner   => $puppet_agent::params::user,
-    group   => $puppet_agent::params::group,
-    mode    => $mode,
-    source  => $source,
-    require => File[$puppet_agent::params::local_packages_dir],
+  # REMIND: redhat/suse with absolute_source
+  # REMIND: debian with absolute_source
+  # REMIND: solaris 10
+  # REMIND: solaris 11 with manage_repo
+  # REMIND: aix
+  # REMIND: darwin
+  # REMIND: suse 11 and PE
+  if $puppet_agent::collection =~ /core/ and $facts['os']['family'] =~ /windows/ {
+    $download_username = getvar('puppet_agent::username', 'forge-key')
+    $download_password = unwrap(getvar('puppet_agent::password'))
+
+    $_download_puppet = windows_native_path("${facts['env_temp_variable']}/download_puppet.ps1")
+    file { $_download_puppet:
+      ensure  => file,
+      content => Sensitive(epp('puppet_agent/download_puppet.ps1.epp')),
+    }
+
+    exec { 'Download Puppet Agent':
+      command  => "${facts['os']['windows']['system32']}\\WindowsPowerShell\\v1.0\\powershell.exe \
+      -ExecutionPolicy Bypass \
+      -NoProfile \
+      -NoLogo \
+      -NonInteractive \
+      ${_download_puppet}",
+      creates  => $local_package_file_path,
+      provider => powershell,
+    }
+  } else {
+    file { $local_package_file_path:
+      ensure  => file,
+      owner   => $puppet_agent::params::user,
+      group   => $puppet_agent::params::group,
+      mode    => $mode,
+      source  => $source,
+      require => File[$puppet_agent::params::local_packages_dir],
```

Comment:
Needs to include this require statement in a windows case, @joshcooper .

Reply:
Updated